doc/spec.md

```diff
@@ -1498,6 +1498,60 @@ a type reference of the form 'G&lt;A, B>' places no requirements on 'A' but requ
 
 The process of substituting type arguments for type parameters in a generic type or generic signature is known as ***instantiating*** the generic type or signature. Instantiation of a generic type or signature can fail if the supplied type arguments do not satisfy the constraints of their corresponding type parameters.
 
+### Indexed access types { #indexed-access-types }
+
+An indexed access type represents the type of a property, even when the type or the property name is unknown.
+
+&emsp;&emsp;*IndexedAccessType:*  
+&emsp;&emsp;&emsp;*Type*&emsp;`[`&emsp;*Type*&emsp;`]`
+
+
+Indexed access types can be created with the above type syntax `T[K]`, as well as the expression syntax `o[k]`.
+In the preceding example, the indexed type is *T* and the access type is *K*.
```

Comment:
I think that instead of "access type", we should call this the "key type" or the "property key type"